packages/gatsby-source-wordpress/src/steps/create-schema-customization/transform-fields/transform-object.js

```diff
@@ -5,13 +5,15 @@ import { getGatsbyApi } from "~/utils/get-gatsby-api"
 import { inPreviewMode } from "~/steps/preview/index"
 import { getPluginOptions } from "../../../utils/get-gatsby-api"
 import { usingGatsbyV4OrGreater } from "~/utils/gatsby-version"
+import { findNamedTypeName, introspectionFieldTypeToSDL } from "../helpers"
 
 export const transformListOfGatsbyNodes = ({ field, fieldName }) => {
-  const typeName = buildTypeName(field.type.ofType.name)
+  const typeSDLString = introspectionFieldTypeToSDL(field.type)
+  const typeName = buildTypeName(findNamedTypeName(field.type))
 
   return {
-    type: `[${typeName}]`,
-    resolve: (source, args, context) => {
+    type: typeSDLString,
```

Comment:
This was a problem as new wpgql types are more like `[${typeName}!]!`. Now this string is generated dynamically for forwards/backwards compat

Reply:
I didn't add this to all type transformers because when I tried that it caused some errors with ID fields going missing. Instead of spending more time fixing it I left it as-is for now. In the future we will be making a new source plugin toolkit and this code will go away anyway.